src/vm/i386/umthunkstub.S

```diff
@@ -173,25 +173,20 @@ LOCAL_LABEL(UMThunkStub_CopyStackArgs):
 LOCAL_LABEL(InitCopyStack):
     push    ecx
     push    edx
-    mov     eax, 0  // Offset in edi
-    mov     ebx, 0  // Offset in esi
-LOCAL_LABEL(CopyStack):
     mov     edx, dword ptr [ebp - UMThunkStub_UMENTRYTHUNK_OFFSET]
     mov     edx, dword ptr [edx + UMEntryThunk__m_pUMThunkMarshInfo]
-    mov     ecx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]
-    cmp     ecx, ebx
+    mov     ebx, [edx + UMThunkMarshInfo__m_cbActualArgSize]
```

Comment:
There is a little bug here, probably induced by my previous comment. You'll need to subtract 4 from ebx before the loop, otherwise the offset comparison below won't work as expected. And change the `mov     ecx, dword ptr [esi + ebx - 4]` to `mov     ecx, dword ptr [esi + ebx]` and the `jnz` at the end to `jnc`.
And if you want, you can move the `add  eax, -4` before the `mov     ecx, dword ptr [esi + ebx]` and then change the `mov     dword ptr [edi + eax - 4], ecx` to `mov     dword ptr [edi + eax], ecx` to make the code look symmetric after the previous change.